inline-r/src/Foreign/R/EventLoop.chs

```diff
@@ -0,0 +1,85 @@
+-- |
+-- Copyright: (C) 2015 Tweag I/O Limited.
+--
+-- Bindings for @<R/R_ext/eventloop.h>@, for building event loops.
+
+{-# LANGUAGE CPP #-}
+{-# LANGUAGE RecordWildCards #-}
+
+module Foreign.R.EventLoop
+  ( InputHandler(..)
+  , inputHandlers
+  , polledEvents
+  , pollingPeriod
+  , graphicsPolledEvents
+  , graphicsPollingPeriod
+  , checkActivity
+  , runHandlers
+  ) where
+
+import Control.Applicative
+import Foreign (FunPtr, Ptr, Storable(..), castPtr)
+import Foreign.C
+import System.Posix.Types (Fd(..))
+import Prelude -- Silence AMP warning.
+
+#include <R_ext/eventloop.h>
+
+-- | R input handler chain. Each input handler points to the next. This view of
+-- input handlers is /shallow/, in the sense that the 'Storable' instance only
+-- unmarshalls the first element in the chain at any one time. A shallow view
+-- allows 'peek' and 'poke' to be inlinable.
+data InputHandler = InputHandler
```

Comment:
Yes - it makes it possible to implement e.g. `registerREvents`, or any other function that wants to enumerate R input handlers to incorporate these handlers into some other event loop iteration function. The alternative is a bunch of accessor functions, but why use that when you can create a cheap view of an R data structure which most of the time gets inlined and simplified away anyways.